app/src/main/java/in/testpress/testpress/util/RegisterFormValidation.kt

```diff
@@ -0,0 +1,105 @@
+package `in`.testpress.testpress.util
+
+import `in`.testpress.testpress.authenticator.RegisterActivity.VerificationMethod
+import `in`.testpress.testpress.databinding.RegisterActivityBinding
+import `in`.testpress.testpress.models.UserDetails
+import android.view.View
+import android.widget.TextView
+import androidx.databinding.BaseObservable
+
+class RegisterFormValidation(
+        private val user: UserDetails,
+        private val binding: RegisterActivityBinding,
+        private val verificationMethod: VerificationMethod,
+        private val isTwilioEnabled: Boolean
+) : BaseObservable() {
+
+    private var isValid = true
+
+    fun isValid(): Boolean {
+        checkAndSetEmptyError()
+        verifyInput()
+        return isValid
+    }
+
+    private fun checkAndSetEmptyError() {
+        if (user.username.trim().isEmpty()) {
+            setErrorText(binding.usernameErrorText, isValid = false)
+        }
+        if (user.password.trim().isEmpty()) {
+            setErrorText(binding.passwordErrorText, isValid = false)
+        }
+        if (user.email.trim().isEmpty()) {
+            setErrorText(binding.emailErrorText, isValid = false)
+        }
+        if (user.confirmPassword.trim().isEmpty()) {
+            setErrorText(binding.confirmPasswordErrorText, isValid = false)
+        }
+
+        if ((user.phoneNumber.trim().isEmpty() && verificationMethod != VerificationMethod.EMAIL)) {
+            setErrorText(binding.passwordErrorText, isValid = false)
+        }
+    }
+
+    private fun verifyInput() {
+        verifyUserName()
+        verifyEmail()
+        if (verificationMethod == VerificationMethod.MOBILE) {
+            verifyPhoneNumber()
+        }
+        verifyPassword()
+        verifyConfirmPassword()
+    }
+
+    private fun verifyUserName() {
```

Comment:
The names aren't explaining what this method is doing. Verify is bit vague. How do you verify? What is the action?

Similarly with other methods.